### PR TITLE
[sc-4443] Toolbar Clean Up

### PR DESCRIPTION
**Pull request by Bondok6**
[Toolbar Clean Up Card](https://app.shortcut.com/helpjuice/story/4443/editor-cleanup)

- Remove the insert media icon from the toolbar
- Re-order the toolbar list
- Decreased the dropdown width for ( Paragraph, font family, and font size ) ( changes in the helpjuice PR)

Helpjuice PR: https://github.com/hajrice/helpjuice4/pull/3172

image

Let me know if you have ideas on how to reduce the toolbar icons further

### CODE REVIEW

**Review by NicholasJarr — COMMENTED** [review]
@Bondok6

I tested and it worked without issues. Just two questions:

1. I used a inline code inserted using the editor on master and inserted the raw HTML on the editor in this branch. The inline code was still persisted in a `code` element, but the styles dropdown didn't show the inline code selected. So for the user to toggle the inline code, he has to click inline code in the editor dropdown twice. Is there a way to make it backward-compatible?
2. The keyboard shortcut doesn't work anymore. Is there a way to add it back?

About any ideas for further improvements, we could:

1. Join all Helpjuice components (tabs, accordions, etc.) in a single element with a dropdown. Not sure if the change is too drastic and might confuse customers though.
2. Make the first section (Paragraph, font size, styles, etc.) more compact. It occupies half of the first row.

**Comment by Bondok6** [comment]
> @Bondok6
> 
> I tested and it worked without issues. Just two questions:
> 
> 1. I used a inline code inserted using the editor on master and inserted the raw HTML on the editor in this branch. The inline code was still persisted in a `code` element, but the styles dropdown didn't show the inline code selected. So for the user to toggle the inline code, he has to click inline code in the editor dropdown twice. Is there a way to make it backward-compatible?

This is a good point, I found that this happens because the styles dropdown is based on the class name, and using inline code through the icon does not generate a class name, it just wraps the text inside a code tag `<code> text </code>`, but using the inline code through the styles dropdown, 
it wraps the text inside a code tag with a class `<code class="fr-text-code"> text </code>`.

So I think to make it works, we'll have to find a way to add this class `fr-text-code` to all inline code generated through the icon.

the second solution I tried but it didn't work, I tried not to use any class in the styles dropdowns and style the text based on the tag name `<code>` but the CK gives me an error saying the classes property is required

I tried this, but it didn't work 
```
	{
		name: "Inline Code",
		element: "code",
		classes: [ ]                  // empty
	}
```

> Now the user can use the remove format button or click the inline code in the editor dropdown twice.

> 2. The keyboard shortcut doesn't work anymore. Is there a way to add it back?

hmm, I think there's no way but I have an idea

Basically keep the inline code feature (I mean don't remove the custom inline code plugin)
I can just remove the icon from the toolbar and keep the functionality ( the same thing I did with the insert media icon )
what do you think?

**Comment by Bondok6** [comment]
> About any ideas for further improvements, we could:
> 
> 1. Join all Helpjuice components (tabs, accordions, etc.) in a single element with a dropdown. Not sure if the change is too drastic and might confuse customers though.
> 2. Make the first section (Paragraph, font size, styles, etc.) more compact. It occupies half of the first row.

Thanks @NicholasJarr for your suggestions
1. Good Idea, but yeh I think this might confuse the customers.
3. I decreased the dropdown width for ( Paragraph, font family, and font size ) in the helpjuice repo

**Comment by NicholasJarr** [comment]
> So I think to make it works, we'll have to find a way to add this class `fr-text-code` to all inline code generated through the icon.

That will solve the problem from now on, but will trigger the same issue on inline code added before theses changes are deployed (we would have to notify the support team about this and that customers will need to remove and re-add the inline code to make it work). I'm not sure if there's an alternative solution besides this one you proposed. @SpaYco any thoughts on this?

> Basically keep the inline code feature (I mean don't remove the custom inline code plugin) I can just remove the icon from the toolbar and keep the functionality ( the same thing I did with the insert media icon ) what do you think?

If we change the feature to add the class as you proposed above, we can go with this 👍🏻 .

> 1. Good Idea, but yeh I think this might confuse the customers

No problem 👍🏻. We can skip this.

> 2. I decreased the dropdown width for ( Paragraph, font family, and font size ) in the helpjuice repo

I checked it on the other PR and it looks really great!

Let's see what @SpaYco has to say about the inline code feature. Besides that, it should be good to go 👍🏻

**Comment by Bondok6** [comment]
Hey @NicholasJarr, 

I tried it and after testing I found that the inline code didn't work as I expected, I need to make more changes to make sure it would work well.

I reverted the changes related to the inline code feature from this PR and added them in another PR https://github.com/Helpjuice/ckeditor5/pull/53 here

So we can safely merge this PR and speed up the process

**Review by NicholasJarr — APPROVED** [review]
> I tried it and after testing I found that the inline code didn't work as I expected, I need to make more changes to make sure it would work well.
> 
> I reverted the changes related to the inline code feature from this PR and added them in another PR #53 here
> 
> So we can safely merge this PR and speed up the process

Sounds good to me 👍🏻. I tested and I noticed no issues.
